rcl_logging_interface/include/rcl_logging_interface/rcl_logging_interface.h

```diff
@@ -91,6 +98,73 @@ RCL_LOGGING_INTERFACE_PUBLIC
 RCUTILS_WARN_UNUSED
 rcl_logging_ret_t rcl_logging_external_set_logger_level(const char * name, int level);
 
+/// Get the logging directory.
+/**
+ * Uses various environment variables to construct a logging directory path.
+ *
+ * Use $ROS2_LOG_DIR if ROS2_LOG_DIR is set and not empty.
+ * Otherwise, use $ROS2_HOME/log, using ~/.ros for ROS2_HOME if not set or if empty.
+ *
+ * It also expands '~' to the current user's home directory.
+ *
+ * \param[in] allocator The allocator to use for memory allocation.
+ * \param[out] directory The C string pointer at which to write the directory path.
+ *   Only meaningful if the call is successful. Must not be null and must point to null.
+ * \return RCL_LOGGING_RET_OK if successful, or
+ * \return RCL_LOGGING_RET_INVALID_ARGUMENT if any arguments are invalid, or
+ * \return RCL_LOGGING_RET_ERROR if an unspecified error occurs.
+ */
+RCL_LOGGING_INTERFACE_PUBLIC
+inline
+rcl_logging_ret_t
+rcl_logging_get_logging_directory(const rcutils_allocator_t * allocator, const char ** directory)
+{
+  if (directory == NULL) {
```

Comment:
nitpick: prefer the literal on the left hand side, to avoid subtle bugs like `if (directory = NULL) {`, see:

https://index.ros.org/doc/ros2/Contributing/Code-Style-Language-Versions/#style

Reply:
ah yeah, I started doing that a while back, but after looking at the `rcl_logging_*` packages I assumed that it wasn't the preferred way. I'll start referring to the style guide directly from now on! Fixed in 269e9018d95a0f5f74bd1c6d642a55442f3e1670